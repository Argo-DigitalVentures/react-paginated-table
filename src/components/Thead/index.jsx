import React, { PropTypes } from 'react';

import ThComponent from '../Th';

import {
  styleType,
  classNameType,
  dataType,
  columnType,
  sortType,
} from '../../propTypes';

function Thead(props) {
  const {
    style,
    className,
    data,
    columns,
    sorts,
    handleHeaderClick,
    ThComponent,
    thStyle,
    thClassName,
    ...rest
  } = props;
  return (
    <thead style={style} className={className} {...rest}>
      <tr>
        {
          columns.map((column, columnIndex, columns) => (
            <ThComponent
              key={column.id}
              style={thStyle}
              className={thClassName}
              column={column}
              columnIndex={columnIndex}
              columns={columns}
              data={data}
              sorts={sorts}
              handleHeaderClick={handleHeaderClick}
            />
          ))
        }
      </tr>
    </thead>
  );
}

Thead.defaultProps = {
  style: {},
  className: '',
  ThComponent,
  thStyle: {},
  thClassName: '',
};

Thead.propTypes = {
  data: dataType.isRequired,
  columns: PropTypes.arrayOf(columnType).isRequired,
  sorts: PropTypes.arrayOf(sortType).isRequired,
  handleHeaderClick: PropTypes.func.isRequired,
  style: PropTypes.object, // eslint-disable-line react/forbid-prop-types
  className: PropTypes.string,
  ThComponent: PropTypes.func,
  thStyle: styleType,
  thClassName: classNameType,
};

export default Thead;
