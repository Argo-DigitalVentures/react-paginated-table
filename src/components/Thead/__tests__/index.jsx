import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Thead from '../';

const data = [
  { company: 'Argo', location: 'New York' },
  { company: 'John Hancock', location: 'New York' },
];

const columns = [
  { id: 'company', header: 'Company Name', selector: 'company' },
  { id: 'location', header: () => null, selector: 'location' },
];

describe('<Thead />', () => {
  it('should accept className, style, and other DOM props', () => {
    const tree = shallow(
      <Thead
        style={{
          color: 'green',
        }}
        className="test"
        data={data}
        columns={columns}
        sorts={[]}
        handleHeaderClick={() => null}
        thStyle={{}}
        thClassName="thTestClassName"
      />,
    );
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('should use a custom ThComponent', () => {
    const FakeThComponent = () => null;
    const tree = shallow(
      <Thead
        style={{
          color: 'green',
        }}
        className="test"
        data={data}
        columns={columns}
        sorts={[]}
        handleHeaderClick={() => null}
        ThComponent={FakeThComponent}
        thStyle={{}}
        thClassName="thTestClassName"
      />,
    );
    expect(toJson(tree)).toMatchSnapshot();
  });
});
