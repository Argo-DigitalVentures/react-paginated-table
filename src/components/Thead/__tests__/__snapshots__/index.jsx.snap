exports[`<Thead /> should accept className, style, and other DOM props 1`] = `
<thead
  className="test"
  style={
    Object {
      "color": "green",
    }
  }>
  <tr>
    <Th
      className="thTestClassName"
      column={
        Object {
          "header": "Company Name",
          "id": "company",
          "selector": "company",
        }
      }
      columnIndex={0}
      columns={
        Array [
          Object {
            "header": "Company Name",
            "id": "company",
            "selector": "company",
          },
          Object {
            "header": [Function],
            "id": "location",
            "selector": "location",
          },
        ]
      }
      data={
        Array [
          Object {
            "company": "Argo",
            "location": "New York",
          },
          Object {
            "company": "John Hancock",
            "location": "New York",
          },
        ]
      }
      handleHeaderClick={[Function]}
      sorts={Array []}
      style={Object {}} />
    <Th
      className="thTestClassName"
      column={
        Object {
          "header": [Function],
          "id": "location",
          "selector": "location",
        }
      }
      columnIndex={1}
      columns={
        Array [
          Object {
            "header": "Company Name",
            "id": "company",
            "selector": "company",
          },
          Object {
            "header": [Function],
            "id": "location",
            "selector": "location",
          },
        ]
      }
      data={
        Array [
          Object {
            "company": "Argo",
            "location": "New York",
          },
          Object {
            "company": "John Hancock",
            "location": "New York",
          },
        ]
      }
      handleHeaderClick={[Function]}
      sorts={Array []}
      style={Object {}} />
  </tr>
</thead>
`;

exports[`<Thead /> should use a custom ThComponent 1`] = `
<thead
  className="test"
  style={
    Object {
      "color": "green",
    }
  }>
  <tr>
    <FakeThComponent
      className="thTestClassName"
      column={
        Object {
          "header": "Company Name",
          "id": "company",
          "selector": "company",
        }
      }
      columnIndex={0}
      columns={
        Array [
          Object {
            "header": "Company Name",
            "id": "company",
            "selector": "company",
          },
          Object {
            "header": [Function],
            "id": "location",
            "selector": "location",
          },
        ]
      }
      data={
        Array [
          Object {
            "company": "Argo",
            "location": "New York",
          },
          Object {
            "company": "John Hancock",
            "location": "New York",
          },
        ]
      }
      handleHeaderClick={[Function]}
      sorts={Array []}
      style={Object {}} />
    <FakeThComponent
      className="thTestClassName"
      column={
        Object {
          "header": [Function],
          "id": "location",
          "selector": "location",
        }
      }
      columnIndex={1}
      columns={
        Array [
          Object {
            "header": "Company Name",
            "id": "company",
            "selector": "company",
          },
          Object {
            "header": [Function],
            "id": "location",
            "selector": "location",
          },
        ]
      }
      data={
        Array [
          Object {
            "company": "Argo",
            "location": "New York",
          },
          Object {
            "company": "John Hancock",
            "location": "New York",
          },
        ]
      }
      handleHeaderClick={[Function]}
      sorts={Array []}
      style={Object {}} />
  </tr>
</thead>
`;
